File: Source/ARTMsgpackEncoder.m
//
//  ARTMsgPackEncoder.m
//  ably
//
//  Created by Toni Cárdenas on 21/3/16.
//  Copyright © 2016 Ably. All rights reserved.
//

#import "ARTMsgPackEncoder.h"
#import "MessagePack.h"

@implementation ARTMsgPackEncoder

- (NSString *)mimeType {
    return @"application/x-msgpack";
}

- (ARTEncoderFormat)format {
    return ARTEncoderFormatMsgPack;
}

- (NSString *)formatAsString {
    return @"msgpack";
}

- (id)decode:(NSData *)data {
    return [data messagePackParse];
}

- (NSData *)encode:(id)obj {
    return [obj messagePack];
}

@end

File: Sources/AblySwift/Core/ARTMsgPackEncoder.swift
import Foundation

public class ARTMsgPackEncoder: ARTJsonLikeEncoderDelegate {
    
    public func mimeType() -> String {
        return "application/x-msgpack"
    }
    
    public func format() -> ARTEncoderFormat {
        return .msgPack
    }
    
    public func formatAsString() -> String {
        return "msgpack"
    }
    
    public func decode(_ data: Data) throws -> Any? {
        // MessagePack decoding - placeholder for now
        // Will need MessagePack framework integration
        fatalError("MessagePack decoding not yet implemented in Swift migration")
    }
    
    public func encode(_ obj: Any) throws -> Data? {
        // MessagePack encoding - placeholder for now
        // Will need MessagePack framework integration  
        fatalError("MessagePack encoding not yet implemented in Swift migration")
    }
}